### PR TITLE
mosquitto: bump

### PR DESCRIPTION
**Pull request by hoanga**
update mosquitto to 2.0.14

### CODE REVIEW

File: app-misc/mosquitto/patches/mosquitto-2.0.14.patchset [hunk 1]
```diff
@@ -0,0 +1,22 @@
+From d74683d456a94b57a806db50824b34391375e1b9 Mon Sep 17 00:00:00 2001
+From: Al Hoang <3811822-hoanga@users.noreply.gitlab.com>
+Date: Sat, 16 Apr 2022 11:32:48 -0500
```
Comment (Begasus) [inline app-misc/mosquitto/patches/mosquitto-2.0.14.patchset:3]: Would be nice if you kept credits to previous patchset creator :)

File: app-misc/mosquitto/patches/mosquitto-2.0.14.patchset [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+     elseif(QNX)
+         set(MOSQ_LIBS ${MOSQ_LIBS} m socket)
++    elseif (${CMAKE_SYSTEM_NAME} MATCHES "Haiku")
++	    set (MOSQ_LIBS ${MOSQ_LIBS} m network)
+     else(APPLE)
+         set (MOSQ_LIBS ${MOSQ_LIBS} dl m)
+         find_library(LIBRT rt)
+-- 
+2.30.2
+
```
Comment (Begasus) [inline app-misc/mosquitto/patches/mosquitto-2.0.14.patchset:22]: Actually the patch isn't needed anymore, seems upstreamed :)

File: app-misc/mosquitto/mosquitto-2.0.14.recipe [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	devel:libmosquittopp$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	mosquitto$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	docbook_xml_dtd$secondaryArchSuffix
+	docbook_xsl_stylesheets$secondaryArchSuffix
```
Comment (Begasus) [inline app-misc/mosquitto/mosquitto-2.0.14.recipe:56]: No secondaryArchSuffix here:
```diff
-	docbook_xml_dtd$secondaryArchSuffix
-	docbook_xsl_stylesheets$secondaryArchSuffix
+	docbook_xml_dtd
+	docbook_xsl_stylesheets
```

**Review by Begasus — CHANGES_REQUESTED** [review]

File: app-misc/mosquitto/patches/mosquitto-2.0.14.patchset (hunk 2 above)
Comment (hoanga) [reply app-misc/mosquitto/patches/mosquitto-2.0.14.patchset]: less patches sounds good to me! thought the patch was needed but looks like not so. i have pushed an update without the patch and looks like it is building from my light testing so far.

and apologies on the earlier patch mangling the author line, i think i  missed the (magical) incantation to update the patch + preserve original author option

File: app-misc/mosquitto/mosquitto-2.0.14.recipe [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+LICENSE="EDL"
+REVISION="1"
+SOURCE_URI="https://mosquitto.org/files/source/mosquitto-$portVersion.tar.gz"
+CHECKSUM_SHA256="d0dde8fdb12caf6e2426b4f28081919a2fce3448773bdb8af0d3cd5fe5776925"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
```
Comment (Begasus) [inline app-misc/mosquitto/mosquitto-2.0.14.recipe:21]: Actually I think this should be:  `libVersionCompat="$libVersion compat >= 1"`

**Comment by Begasus** [comment]
Same here, not used by anything from what I can tell, remove the old recipe also?

**Review by Begasus — APPROVED** [review]

File: app-misc/mosquitto/mosquitto-2.0.14.recipe (hunk 4 above)
Comment (korli) [reply app-misc/mosquitto/mosquitto-2.0.14.recipe:21]: I agree

**Comment by korli** [comment]
Please add optional dependencies c-ares and cjson, both should be available.

File: app-misc/mosquitto/mosquitto-2.0.14.recipe (hunk 4 above)
Comment (hoanga) [reply app-misc/mosquitto/mosquitto-2.0.14.recipe]: updated per suggestion

**Comment by hoanga** [comment]
i have added dependencies for c-ares but could not find an existing port for cjson to use. the closest i can find is json_c which i tried to use (unsuccessfully).  i have only updated for c-ares for now. as for removing the older version, is it preferred in a separate pr?

File: app-misc/mosquitto/mosquitto-2.0.14.recipe [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	cmd:mosquitto_passwd$secondaryArchSuffix = $portVersion
+	cmd:mosquitto_pub$secondaryArchSuffix = $portVersion
+	cmd:mosquitto_rr$secondaryArchSuffix = $portVersion
+	cmd:mosquitto_sub$secondaryArchSuffix = $portVersion
+	lib:libmosquitto$secondaryArchSuffix = $libVersionCompat
+	lib:libmosquittopp$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	c_ares
```
Comment (korli) [inline app-misc/mosquitto/mosquitto-2.0.14.recipe:39]:
```suggestion
	lib:libcares$secondaryArchSuffix
```

File: app-misc/mosquitto/mosquitto-2.0.14.recipe [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	"
+REQUIRES_devel="
+	mosquitto$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	docbook_xml_dtd
+	docbook_xsl_stylesheets
+	c_ares
```
Comment (korli) [inline app-misc/mosquitto/mosquitto-2.0.14.recipe:57]:
```suggestion
	devel:libcares$secondaryArchSuffix
```

**Comment by korli** [comment]
thanks. indeed cjson isn't yet available at haikuports.

**Pull request merged**